dmr/src/main/java/org/jboss/as/console/client/shared/state/WarningProcessor.java

```diff
@@ -0,0 +1,60 @@
+package org.jboss.as.console.client.shared.state;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+
+import org.jboss.dmr.client.ModelNode;
+import org.jboss.dmr.client.Property;
+import org.jboss.dmr.client.notify.Notifications;
+
+public class WarningProcessor implements ResponseProcessor {
+
+    protected static final String LEVEL = "level";
+    protected static final String RESPONSE_HEADERS = "response-headers";
+    protected static final String RESULT = "result";
+    protected static final String STEP_1 = "step-1";
+    protected static final String WARNING = "warning";
+    protected static final String WARNINGS = "warnings";
+
+    @Override
+    public boolean accepts(ModelNode response) {
+        // result->step-x->response-headers->warnings
+        if (response.hasDefined(RESULT)) {
```

Comment:
Right, but that might change - and there might be other operations which cause these response headers. So it would be more robust to be prepared for warnings which are not wrapped inside steps.